src/Symfony/Bundle/SecurityBundle/Command/UserPasswordEncoderCommand.php

```diff
@@ -150,13 +155,9 @@ private function createSaltQuestion(InputInterface $input, OutputInterface $outp
     {
         $saltQuestion = new Question("\n > (Optional) <question>Provide a salt (press <enter> to generate one):</question> ");
 
-        $container = $this->getContainer();
-        $saltQuestion->setValidator(function ($value) use ($output, $container) {
+        $saltQuestion->setValidator(function ($value) use ($output) {
             if ('' === trim($value)) {
-                $value = hash('sha512', $container->get('security.secure_random')->nextBytes(30));
-
-                $output->writeln("\n<comment>The salt has been generated: </comment>".$value);
-                $output->writeln(sprintf("<comment>Make sure that your salt storage field fits this salt length: %s chars.</comment>\n", strlen($value)));
+                $value = $this->generateSalt($output);
```

Comment:
It is => https://github.com/symfony/symfony/blob/2.7/src/Symfony/Bundle/SecurityBundle/Tests/Functional/UserPasswordEncoderCommandTest.php